pkg/destroy/gcp/gcp.go

```diff
@@ -40,6 +42,14 @@ const (
 	// capgProviderOwnedLabelFmt is the format string for the label
 	// used for resources created by the Cluster API GCP provider.
 	capgProviderOwnedLabelFmt = "capg-cluster-%s"
+
+	// gcpGlobalResource is an identifier to indicate that the resource(s)
+	// that are being deleted are globally scoped.
+	gcpGlobalResource resourceScope = "global"
+
+	// gcpRegionalResource is an identifier to indicate that the resource(s)
+	// that are being deleted are regionally scoped.
+	gcpRegionalResource resourceScope = "regional"
```

Comment:
Instead of switching on global vs regional, perhaps we just always try to delete both? 

We generally need to be careful about how we delete resources, but if something is tagged with the infraID for the cluster it should be safe to destroy.

Reply:
@patrickdillon If there is a way to delete both global and regional resources with a single call that would work. As far as I am aware there is not, so the deletion requires finding then deleting global or finding then deleting regional resources. This is the reason that I have opted to create these identifiers.